Add documentation for GitHub App API changes within Terraform Cloud (#283)
hashicorp/terraform-docs-common: website/docs/cloud-docs/api-docs/private-registry/modules.mdx

| --------------------------------------------- | ------- | ------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `data.type`                                   | string  |         | Must be `"registry-modules"`.                                                                                                                                    |
| `data.attributes.vcs-repo.identifier`         | string  |         | The repository from which to ingress the configuration.                                                                                                          |

> | `data.attributes.vcs-repo.oauth-token-id`     | string  |         | The VCS Connection (OAuth Connection + Token) to use as identified. You can obtain this ID from the [oauth-tokens](/terraform/cloud-docs/api-docs/oauth-tokens) endpoint. This value must not be specified if `github-app-installation-id` is specified. |

Reviewer: I'm not sure what `to use as identified` means here so I didn't make a specific suggestion, but I think that could use some clarifying. 

```suggestion
| `data.attributes.vcs-repo.oauth-token-id`     | string  |         | The VCS Connection (OAuth Connection + Token) to use as identified. Get this ID from the [oauth-tokens](/terraform/cloud-docs/api-docs/oauth-tokens) endpoint. You can not specify this value if `github-app-installation-id` is specified. |
```

> | `data.attributes.vcs-repo.github-app-installation-id` | string |  | The VCS Connection GitHub App Installation to use. This ID can be obtained from the user settings page. Requires previously authorizing the GitHub App and generating a user-to-server token. You can manage this from User Settings within Terraform Cloud. This value must not be specified if `oauth-token-id` is specified.                                                                                                                                                      |

Reviewer: ```suggestion
| `data.attributes.vcs-repo.github-app-installation-id` | string |  | The VCS Connection GitHub App Installation to use. Find this ID on the user settings page. Requires previously authorizing the GitHub App and generating a user-to-server token. Manage the token from **User Settings** within Terraform Cloud. You can not specify this value if `oauth-token-id` is specified.                                                                                                                                                      |
```
